File: app/src/main/java/software/credible/abercrombiefitchkata/AnfPromoApplication.java
package software.credible.abercrombiefitchkata;

import android.app.Application;
import android.content.Intent;

import io.realm.Realm;
import io.realm.RealmConfiguration;
import software.credible.abercrombiefitchkata.remote.FetchPromosIntentService;

public class AnfPromoApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        initRealmWithDefaultConfig();
        schedulePromoFetch();
    }

    private void initRealmWithDefaultConfig() {
        Realm.init(this);
        Realm.setDefaultConfiguration(
                new RealmConfiguration.Builder()
                        .deleteRealmIfMigrationNeeded()
                        .build());
    }

    private void schedulePromoFetch() {
        Intent intent = new Intent(this, FetchPromosIntentService.class);
        startService(intent);
    }

}

File: app/src/main/java/software/credible/abercrombiefitchkata/AnfPromoApplication.kt
package software.credible.abercrombiefitchkata

import android.app.Application
import android.content.Intent

import io.realm.Realm
import io.realm.RealmConfiguration
import software.credible.abercrombiefitchkata.remote.FetchPromosIntentService

class AnfPromoApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        initRealmWithDefaultConfig()
        schedulePromoFetch()
    }

    private fun initRealmWithDefaultConfig() {
        Realm.init(this)
        Realm.setDefaultConfiguration(
                RealmConfiguration.Builder()
                        .deleteRealmIfMigrationNeeded()
                        .build())
    }

    private fun schedulePromoFetch() {
        val intent = Intent(this, FetchPromosIntentService::class.java)
        startService(intent)
    }

}
